appscale/tools/remote_helper.py

```diff
@@ -151,27 +153,72 @@ def start_all_nodes(cls, options, node_layout):
 
     agent.configure_instance_security(params)
 
-    load_balancer_nodes = node_layout.get_nodes('load_balancer', True)
-    instance_ids, public_ips, private_ips = cls.spawn_load_balancers_in_cloud(
-      options, agent, params,
-      len(load_balancer_nodes))
+    load_balancer_roles = {}
+    instance_type_roles = {}
 
-    for node_index, node in enumerate(load_balancer_nodes):
-      index = node_layout.nodes.index(node)
-      node_layout.nodes[index].public_ip = public_ips[node_index]
-      node_layout.nodes[index].private_ip = private_ips[node_index]
-      node_layout.nodes[index].instance_id = instance_ids[node_index]
+    for node in node_layout.get_nodes('load_balancer', True):
+      load_balancer_roles.setdefault(node.instance_type, []).append(node)
+
+    for node in node_layout.get_nodes('load_balancer', False):
+      instance_type = instance_type_roles
+      instance_type.setdefault(node.instance_type, []).append(node)
+
+    spawned_instance_ids = []
+
+    for instance_type, load_balancer_nodes in load_balancer_roles.items():
+      # Copy parameters so we can modify the instance type.
+      instance_type_params = params.copy()
+      instance_type_params['instance_type'] = instance_type
+
+      instance_ids, public_ips, private_ips = cls.spawn_nodes_in_cloud(
```

Comment:
Does this need to clean up after itself if it encounters an exception?

Reply:
Yes it does, It left my mind that we could technically have different instance types for LBs and it would need clean up if the 2nd LB faced a problem after the 1st one was started successfully. I have added the same try except as for non load balancer nodes.